### PR TITLE
Moving away from novaclient and adding all the missing pieces into reddw...

### PR DESCRIPTION
**Pull request by rnirmal**
...arfclient.
- Added parameters for the authentication instead of the arguments.
- Cleaned out the HttpClient and Authentication pieces.

### CODE REVIEW

**Comment by rnirmal** [comment]
This sill needs some refining, just putting out there for everyone to review. Need to make sure the named arguments are ok, proper help messages on those etc etc...

**Comment by hub-cap** [comment]
so by doing this, we have ot fix it every time something like keystone changes. I just want to say that i understand why we are doing this but im not a fan at all.

**Comment by TimSimpsonR** [comment]
We give up the ability of having keystone fixes automatically given to us. We gain the ability of no longer getting broken by nova-client changes we weren't expecting. Additionally even if nova-client hits pypi I think the lack of dependencies will be appreciated.

File: reddwarfclient/base.py [hunk 1]
```diff
@@ -1,14 +1,293 @@
-def isid(obj):
+# Copyright 2010 Jacob Kaplan-Moss
```
Comment (ed-) [inline reddwarfclient/base.py:1]: oh.

**Comment by TimSimpsonR** [comment]
+1

**Comment by ed-** [comment]
+1

**Pull request merged**